src/core/reference/include/openvino/reference/rms_norm.hpp

```diff
@@ -72,5 +73,33 @@ void rms_norm(const T* in,
     rms_norm(in, axes, out, in_shape, eps);
     multiply(out, scale, out, in_shape, scale_shape, op::AutoBroadcastType::NUMPY);
 }
+
+/**
+ * @brief Reference implementation of RMS operator with output type conversion
+ *
+ *  Math Formula: Convert((x / Sqrt(ReduceMean(x^2, axes) + eps)) * scale), T_OUT)
+ *
+ * @param in           Input pointer to data
+ * @param axes         Axes for reduce mean calculation
+ * @param out          Output pointer to results
+ * @param in_shape     Shape of the input Tensor
+ * @param eps          Epsilon for not dividing by zero while normalizing the value
+ * @param scale_shape  Shape of the scale Tensor
+ * @param scale        Input pointer to scale
+ *
+ */
+template <class T_IN, class T_OUT>
+void rms_norm_mul_convert_out(const T_IN* in,
+                              const AxisSet& axes,
+                              T_OUT* out,
+                              const Shape& in_shape,
+                              double eps,
+                              const Shape& scale_shape,
+                              const T_IN* scale) {
+    std::vector<T_IN> tmp_out(shape_size(in_shape));
+    rms_norm(in, axes, tmp_out.data(), in_shape, eps, scale_shape, scale);
+    convert(tmp_out.data(), out, tmp_out.size());
```

Comment:
```suggestion
    rms_norm(in, axes, out, in_shape, eps, scale_shape, scale);
    convert(out, out, tmp_out.size());
```
The `out_temp` should not be required.

Reply:
Could you please elaborate a little bit more? 

The goal is to convert f32 to f16 (when input data is f32 and output f16), the proposed change `convert(out, out, ...);` seems to have no effect.
The conversion should be optimized in plugins for sure, but here the other reference functions like `power` or `reduce_mean` inside the `rms_norm` require the in/out ptr to have the same type. 
So seems like the temp vector will be needed at some point of the ref impl anyway.

Reply:
The suggestion was wrong. But if `T_IN and T_OUT` are same type the additional buffer and convert (dummy copy) are not required.
The template could have special version for such case.

Reply:
The `rms_norm` can be used if conversion is not needed. It expects `T` in/out of the same type.

https://github.com/openvinotoolkit/openvino/blob/10bc58263c756026690c843b32e5dbf1a255621f/src/core/reference/include/openvino/reference/rms_norm.hpp#L32-L33

https://github.com/openvinotoolkit/openvino/blob/10bc58263c756026690c843b32e5dbf1a255621f/src/core/reference/include/openvino/reference/rms_norm.hpp#L65-L72